lib/galaxy/datatypes/tabular.py

```diff
@@ -1297,3 +1297,82 @@ def set_meta(self, dataset, overwrite=True, skip=None, max_data_lines=5, **kwd):
+
+                        if l.startswith('#h'):
+
+                            column_headers = l.strip('\n').split("\t")[1:]
+                        elif l.startswith('#f'):
+                            cleaned_column_types = []
+                            for i in l.split('\t')[1:]:
+                                if i.strip("\n") == 'Hex':
+                                    cleaned_column_types.append('str')
+                                else:
+                                    cleaned_column_types.append(i)
+                        comment_lines += 1
+                        fields = l.split('\t')
+                        if len(fields) == 2:
+                            if fields[0] == '# CMAP File Version:':
+                                dataset.metadata.cmap_version = fields[1].strip('\n')
+                            elif fields[0] == '# Label Channels:':
+                                dataset.metadata.label_channels = int(fields[1].strip('\n'))
+                            elif fields[0] == '# Nickase Recognition Site 1:':
+                                fields2 = fields[1].strip('\n').split(';')
+                                if len(fields2) == 2:
+                                    dataset.metadata.channel_1_color = fields2[1]
+                                dataset.metadata.nickase_recognition_site_1 = fields2[0].split(',')
+                            elif fields[0] == '# Number of Consensus Maps:':
+                                dataset.metadata.number_of_consensus_nanomaps = int(fields[1].strip('\n'))
+                            elif fields[0] == '# Nickase Recognition Site 2:':
+                                fields2 = fields[1].strip('\n').split(';')
+                                if len(fields2) == 2:
+                                    dataset.metadata.channel_2_color = fields2[1]
+                                dataset.metadata.nickase_recognition_site_2 = fields2[0].split(',')
+                    elif self.max_optional_metadata_filesize >= 0 and dataset.get_size() > self.max_optional_metadata_filesize:
+                        # If the dataset is larger than optional_metadata, just count comment lines.
+                        # No more comments, and the file is too big to look at the whole thing. Give up.
+                        dataset.metadata.data_lines = None
+                        break
+                    elif i == comment_lines + 1:
+                        number_of_columns = int(len(l.split('\t')))
+                if not (self.max_optional_metadata_filesize >= 0 and dataset.get_size() > self.max_optional_metadata_filesize):
+                    dataset.metadata.data_lines = i + 1 - comment_lines
+            dataset.metadata.comment_lines = int(comment_lines)
+            dataset.metadata.column_names = column_headers
+            dataset.metadata.column_types = cleaned_column_types
+            dataset.metadata.columns = int(number_of_columns)
```

Comment:
```suggestion
            dataset.metadata.columns = number_of_columns
```
same: already an int